### PR TITLE
Use parent directory of setup.py path

### PR DESCRIPTION
**Pull request by goodspark**
If a package is nested in a sub directory, then this plugin doens't
work. So change the working directory based on the `setupfile` path
(with the code to restore the original directory afterwards, just in
case).

### CODE REVIEW

File: plugin.go [hunk 1]
```diff
@@ -2,7 +2,9 @@ package main
```
Comment (goodspark) [inline plugin.go]: This CI error looks unrelated to my changes.

```
Installing collected packages: setuptools, wheel, pycparser, cffi
86 | Running setup.py install for pycparser: started
87 | Running setup.py install for pycparser: finished with status 'done'
88 | Running setup.py install for cffi: started
89 | Running setup.py install for cffi: finished with status 'error'
90 | ERROR: Command errored out with exit status 1:
91 | command: /usr/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-_w8c5tvi/cffi/setup.py'"'"'; __file__='"'"'/tmp/pip-install-_w8c5tvi/cffi/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-mkly6033/install-record.txt --single-version-externally-managed --prefix /tmp/pip-build-env-80v9kp8z/overlay --compile
92 | cwd: /tmp/pip-install-_w8c5tvi/cffi/
93 | Complete output (48 lines):
94 | unable to execute 'gcc': No such file or directory
95 | unable to execute 'gcc': No such file or directory
```
https://cloud.drone.io/drone-plugins/drone-pypi/49/1/4

File: plugin.go [hunk 2]
```diff
@@ -21,7 +23,15 @@ func (p Plugin) buildCommand() *exec.Cmd {
 	// Set the default of distributions in here
 	// as CLI package still has issues with string slice defaults
 	distributions := []string{"sdist"}
-	if len(p.Distributions) > 0 {
+	// Sanitize bad dist values
+	inputDists := make([]string, 0, len(p.Distributions))
```
Comment (tboerger) [inline plugin.go]: This is really unrelated to the issue and beside that I'm not sure if I would like to manipulate that list like that.

File: plugin.go [hunk 3]
```diff
@@ -46,7 +56,27 @@ func (p Plugin) uploadCommand() *exec.Cmd {
 }
 
 // Exec runs the plugin - doing the necessary setup.py modifications
-func (p Plugin) Exec() error {
+func (p *Plugin) Exec() error {
+	// If a setup.py is in a subdirectory, we need to change to that directory first
+	// so the correct files are packaged.
+	pathParts := strings.Split(p.SetupFile, string(os.PathSeparator))
```
Comment (tboerger) [inline plugin.go]: What about just using pathname functions?

**Comment by tboerger** [comment]
As you can define the dist dir now I think we can close this pull request, right?
